app/controllers/article_controller.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class ArticleController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @articles = Article
+                .where(["user_id = ?", params[:id] || current_user.id])
+                .eager_load(:user)
+                .reorder(created_at: :desc)
+                .all
+    @show_article_author = false
+  end
+
+  def new
+    if request.request_method == "GET"
+      @article = Article.new
+    else
+      @article = Article.create_new permitted_article_fields, current_user
+      if @article.save
+        flash[:success] = "Article created"
+        redirect_to action: :index
+      end
+    end
+  end
+
+  def edit
+    find_article
+
+    if request.request_method == "PATCH"
+      @article.update permitted_article_fields
+      if @article.save
+        flash[:success] = "Article edited"
+        return redirect_to self_articles_list_path
+      end
+    end
+
+    render "article/new"
+  end
+
+  def remove
+    @article = Article.find_by(id: params[:id], user: current_user)
+    @article&.destroy
+  end
+
+  protected
+
+  def find_article
+    @article = Article.find_by(id: params[:id], user: current_user)
+    raise ActionController::RoutingError, "Not Found" unless @article
+  end
+
+  def permitted_article_fields
+    params.require(:article).permit(:message)
```

Comment:
You can add current user right here
```suggestion
    params.require(:article).permit(:message).merge(user: current_user)
```